apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.schema.ts

```diff
@@ -0,0 +1,72 @@
+import { FormikHelpers } from 'formik'
+import * as yup from 'yup'
+
+import { AddressType } from 'src/typings'
+import { addressValidationSchema } from 'src/utils/yup'
+
+export interface Milestone {
+  amount: number
+  title: string
+  endDate: number & string & Date
+  mediaUrl: string | undefined
+  mediaType: string | undefined
+  mediaFileName: string
+  description: string
+}
+
+export interface EscrowFormValues {
+  clientAddress: string | AddressType
+  recipientAddress: string | AddressType
+  safetyValveDate: Date | number | string
+  milestones: Array<Milestone>
+}
+
+export interface EscrowFormState {
+  formValues: EscrowFormValues
+  setFormValues: (values: EscrowFormValues) => void
+  resetForm: () => void
+}
+
+export interface EscrowFormProps {
+  onSubmit: (values: EscrowFormValues, actions: FormikHelpers<EscrowFormValues>) => void
+  isFormSubmitting: boolean
+  escrowFormDataIpfsCID: string
+}
+
+export const MilestoneSchema = yup.object({
+  amount: yup
+    .number()
+    .min(0.1, 'Amount must be greater than 0.1 ETH')
+    .required('Amount is required'),
+  title: yup.string().required('Title is required'),
+  endDate: yup.date().required('End date is required'),
+  mediaUrl: yup.string(),
+  mediaType: yup
+    .string()
+    .oneOf([
+      'image/jpeg',
+      'image/png',
+      'image/svg+xml',
+      'image/webp',
+      'image/gif',
+      'video/mp4',
+      'video/quicktime',
+      'audio/mpeg',
+      'audio/wav',
+      undefined,
+    ]),
+  mediaFileName: yup.string(),
+  description: yup.string(),
+})
+
+export const EscrowFormSchema = yup.object({
+  clientAddress: addressValidationSchema,
+  recipientAddress: addressValidationSchema,
+  safetyValveDate: yup
+    .date()
+    .required('Safety valve date must be at least 30 days from today.'),
```

Comment:
Could we make a check to ensure its atleast 30 days after the enddate of the last milestone ?